ietf/draft-ietf-jmap-calendars-18.txt

```diff
@@ -0,0 +1,3360 @@
+       ...
+     }, "0"]
+   ]
+
+8.3.  Snoozing an alert
+
+   The client is connected to the event source and receives a push:
+
+   {
+           "@type": "CalendarAlert",
+           "accountId": "a0x9",
+           "calendarEventId": "E-7e93e3ee-4e6e-408a-9adc-cbaf1dbd0a3f",
+           "uid": "b6f7e27b-5872-4b52-b457-0242541bb01c",
+           "recurrenceId": null,
+           "alertId": "7519a951-1e6f-4a6c-b08b-20dd2e5a89cd"
+   }
+
+
+
+
+Jenkins & Douglass       Expires 9 December 2024               [Page 45]
+
+Internet-Draft               JMAP Calendars                    June 2024
+
+
+   Not finding this event in its local cache, the client fetches the
+   information for this event that it needs to show the alert by making
+   the following request:
+
+   [
+     ["CalendarEvent/get", {
+           "accountId": "a0x9",
+           "ids":["E-7e93e3ee-4e6e-408a-9adc-cbaf1dbd0a3f"],
+           "properties": ["calendarIds", "title", "start",
+             "timeZone", "useDefaultAlerts", "alerts"]
+     }, "0"]
+   ]
+
+   In response it receives:
+
+   [
+     ["CalendarEvent/get", {
+           "accountId": "a0x9",
+           "list": [{
+          "id": "E-01c9626e-1490-43df-a34f-457021256281",
+          "calendarIds": {
+             "3ddf2ad7-0e0c-4fb5-852d-f0ff56f3c662": true
+          },
+          "title": "Team catchup",
+          "start": "20230210T17:00:00",
+          "timeZone": "America/New_York",
+          "useDefaultAlerts": false,
+          "alerts": {
+            "7519a951-1e6f-4a6c-b08b-20dd2e5a89cd": {
+              "@type": "Alert",
+              "action": "display",
+              "trigger": {
+                "@type": "OffsetTrigger",
+                "relativeTo": "start",
+                "offset": "-PT1H"
+              }
+            }
+          }
+        }],
+        "notFound": [],
+        "state": "$$/414/208"
+     }, "0"]
+   ]
+
+   The client displays an alert in a platform-appropriate manner.
+   Presuming the user here is in the Australia/Melbourne time zone, this
+   might look something like:
+
+
+
+
+Jenkins & Douglass       Expires 9 December 2024               [Page 46]
+
+Internet-Draft               JMAP Calendars                    June 2024
+
+
+   +----------------------------------------+
+   |                                        |
+   |  Reminder: Team catchup                |
+   |  Today at 10am (in 1 hour)             |
+   |                             [Snooze\/] |
+   +----------------------------------------+
+
+   The user snoozes the notification for 30 minutes.  The client
+   dismisses the current notification and sends an update to the event
+   to the server:
+
+   NOTE: '\' line wrapping per RFC 8792
+
+   [
+     ["CalendarEvent/set", {
+           "accountId": "a0x9",
+           "update": {
+         "E-01c9626e-1490-43df-a34f-457021256281": {
+           "alerts/7519a951-1e6f-4a6c-b08b-20dd2e5a89cd\
+                   /acknowledged": "20230110T23:00:31Z",
```

Comment:
This acknowledgement is in January, 1 month before the alert fires

Reply:
Thanks, fixed.